extensions/amp-video/0.1/mixins.js

```diff
@@ -0,0 +1,403 @@
+  /** */
+  fullscreenExit() {}
+
+  /** @return {boolean} */
+  isFullscreen() {}
+
+  /**
+   * Toggles video's mute state.
+   * @param {boolean} unusedMuted
+   */
+  getCurrentTime() {}
+
+  /** @return {number} */
+  getDuration() {}
+
+  /** @return {!Array<!Array<number>>} */
+  getPlayedRanges() {}
+}
+
+
+/** @implements {!AmpVideoMixin} */
+export class VideoElementMixin {
+  /**
+   * @param {!./amp-video.AmpVideo} impl
+   */
+  constructor(impl) {
+    /** @protected @const {!./amp-video.AmpVideo} */
+    // TODO(alanorozco): Make private
+    this.impl = impl;
+
+    /** @protected {?HTMLMediaElement} */
+    // TODO(alanorozco): Make private
+    this.video = null;
+  }
+
+  isVideoSupported() {
+    return !!this.video.play;
+  }
+
+  /** @override */
+  createBaseNode() {
+    const {element} = this.impl;
+    const {ownerDocument} = element;
+    this.video = ownerDocument.createElement('video');
+    this.impl.getVsync().mutate(() => {
+      element.appendChild(this.video);
+    });
+    return this.video;
+  }
+
+  /** @override */
+  installEventHandlers(eventsToForward, setMuted) {
+    this.impl.forwardEvents(eventsToForward, this.video);
+
+    listen(this.video, 'volumechange', e => {
+      setMuted(this.video.muted);
+    });
+  }
+
+  /** @override */
+  appendSources(fragment) {
+    this.video.appendChild(fragment);
+  }
+
+  /** @override */
+  whenLoadStarts() {
+    // loadPromise for media elements listens to `loadstart`
+    return this.impl.loadPromise(this.video);
+  }
+
+  /** @override */
+  play() {
+    new Promise(() => this.video.play()).catch(() => {
+      // Empty catch to prevent useless unhandled promise rejection logging.
+      // Play can fail for many reasons such as video getting paused before
+      // play() is finished.
+      // We use events to know the state of the video and do not care about
+      // the success or failure of the play()'s returned promise.
+    });
+  }
+
+  /** @override */
+  pause() {
+    this.video.pause();
+  }
+
+  /** @override */
+  toggleMuted(muted) {
+    this.video.muted = muted;
+  }
+
+  /** @override */
+  toggleControls(show) {
+    this.video.controls = show;
+  }
+
+  /**
+   * @override
+   */
+  fullscreenEnter() {
+    fullscreenEnter(dev().assertElement(this.video_));
+  }
+
+  /**
+   * @override
+   */
+  fullscreenExit() {
+    fullscreenExit(dev().assertElement(this.video_));
+  }
+
+  /** @override */
+  isFullscreen() {
+    return isFullscreenElement(dev().assertElement(this.video_));
+  }
+
+  /** @override */
+  getCurrentTime() {
+    return this.video.currentTime;
+  }
+
+  /** @override */
+  getDuration() {
+    return this.video.duration;
+  }
+
+  /** @override */
+  getPlayedRanges() {
+    return VideoUtils.getPlayedRanges(this.video);
+  }
+}
+
+
+/** @implements {!AmpVideoMixin} */
+// TODO(alanorozco): This mixin should not inherit from `VideoElementMixin`.
+// This is blocked by mediapool control coming from `AmpStoryPage`.
```

Comment:
What does this mean?  How can we unblock this?

Reply:
Sorry, wrote the comment on a rush.

I'd like the registration API to change to avoid appending a video node (similar to [your comment below.](https://github.com/ampproject/amphtml/pull/14316#discussion_r178315703). I made that change for this PR initially, but it changes the semantics of `MediaPool` somewhat, so it became too large a change to review. 

Once that API change is in, the mixin would take care of registering itself in the pool and then won't create/append a node and can then just not inherit from `VideoElementMixin`.